stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -214,6 +226,28 @@ private static StatusRuntimeException toStatusRuntimeException(Throwable t) {
     return Status.UNKNOWN.withCause(t).asRuntimeException();
   }
 
+  /**
+   * Cancels a call, and throws the exception.
+   * 
+   * @param t must be a RuntimeException or Error
+   */
+  private static RuntimeException cancelThrow(ClientCall<?, ?> call, Throwable t) {
+    try {
+      call.cancel(null, t);
+    } catch (RuntimeException e) {
+      logger.log(Level.SEVERE, "RuntimeException encountered while closing call", e);
```

Comment:
Why not just catch Throwable? These code paths are basically the same.

Reply:
I was hoping to (eventually) remove all the catch(throwable) in our code, and limit it only to the exceptions that can actually be thrown.  I can put it back the way it was if youd like, but it makes it harder to catch places where we do it wrong.  For example, we were wrapping Error above, previously.

Reply:
Previously we weren't try-catching around cancel(). That honestly still sounds fine, but if you want the try-catch... sure. But I don't see any need for catching each type individually here.

Reply:
Done, but added an assert comment.